### PR TITLE
[expo-dev-menu-interface] Add auto-linking support

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

We want to make sure that the installation process for `dev-menu` and `dev-client` is as trivial as possible.  So I've added the auto-linking support into `expo-dev-menu-interface`.

# How

- react-native auto-linking works only if the dependency is listed in the package.json. So I've updated the documentation - users need to install both packages `expo-dev-menu` and `expo-dev-menu-interface`.
- create dummy `ReactPackage` - otherwise, the auto-linking doesn't work.
- https://github.com/expo/expo/blob/9262d54309ab52d6b0cdcc60aad9720a9783ccdd/packages/expo-dev-menu-interface/react-native.config.js#L1-L13

# Test Plan

- bare-expo
	- iOS ✅
	- android ✅

### CODE REVIEW

File: apps/bare-expo/package.json [hunk 1]
```diff
@@ -88,6 +88,7 @@
     "@react-native-community/viewpager": "4.1.6",
     "expo": "~39.0.1",
     "expo-dev-menu": "0.0.2",
+    "expo-dev-menu-interface": "0.0.2",
```
Comment (tsapeta) [inline apps/bare-expo/package.json:91]: That makes me a little bit sad too 😿
Would be great if we could force autolinking to take this dependency automatically. Could you do some research how hard would it be to add sth like this to react native cli?

**Review by tsapeta — CHANGES_REQUESTED** [review]

File: apps/bare-expo/ios/BareExpo/AppDelegate.swift [hunk 2]
```diff
@@ -7,7 +7,7 @@
 //
 
 import Foundation
-import EXDevMenuInterface
+import expo_dev_menu_interface
```
[the remark this answers is not in the document]
Comment (lukmccall) [reply apps/bare-expo/ios/BareExpo/AppDelegate.swift]: I changed that ;)

File: apps/bare-expo/package.json (hunk 1 above)
Comment (lukmccall) [reply apps/bare-expo/package.json:91]: I'll do some research ;)

**Review by nikki93 — APPROVED** [review]
Looks fine to me! Did it turn out that the mention in package.json is needed or not? (Honestly to me it's fine either way: package.json is better than the native configs)

**Comment by lukmccall** [comment]
> Did it turn out that the mention in package.json is needed or not? 

Unfortunately, it's needed for now :/

**Comment by nikki93** [comment]
> > Did it turn out that the mention in package.json is needed or not?
> 
> Unfortunately, it's needed for now :/

I see. From my side I'm ok with it bc. as far as dev client goes we may actually just add expo-dev-menu and expo-dev-menu-interface as dependencies for the expo-development-client package, so it could still come down to one line in package.json for users.

**Review by tsapeta — APPROVED** [review]

**Pull request merged**